### PR TITLE
General Performance Improvements

### PR DESCRIPTION
**Pull request by markhalonen**
These are low-risk changes that improve performance. I made Front/Back Face Force lazily calculated, and added some `parallel()` calls where they empirically improved performance.

# How To Test
- Load Luke's workspace that has performance issues and observe speed improvements.
   - The reducer modifier is the "silver bullet" and will be in another PR.

This PR also contains random cleanup/comments and whatnot, may as well review that as well.

### CODE REVIEW

File: Viewer/src/net/relinc/viewer/application/ScaledResults.java [hunk 1]
```diff
@@ -17,20 +17,18 @@
 	private double[] strainRate;
 	private double[] frontFaceForce;
 	private double[] backFaceForce;
-
+	private Sample sample;
 
 	public ScaledResults(Sample s) {
+		this.sample = s;
```
Comment (markhalonen) [inline Viewer/src/net/relinc/viewer/application/ScaledResults.java]: This is a tiny bit risky.. The Sample object could change under our feet between calls to the `ScaledResults` constructor and `getFrontFaceForce`. But the risk seems pretty low. It seemed to work for me. So this optimization is not totally "free" in terms of keeping the behavior of the program the same

**Comment by bhalonen** [comment]
I like this approach, however, I think the --major-- operation is filtering. That is murder.

I'll pull and test this when I have a chance, but I've worked ~11 hours a day at SGR last two days.

**Comment by bhalonen** [comment]
Here is the issue I was concerned about. 
As you can see, the filter is not getting applied to the front face force
image

**Comment by markhalonen** [comment]
You'd have to use the "Global Displacement Filter" to filter the Front Face Force. You are using "Global Load Filter", which only filters Back Face Force. I checked this against master and they behave the same.

File: Viewer/src/net/relinc/viewer/GUI/HomeController.java [hunk 2]
```diff
@@ -874,7 +876,18 @@ public void applySession(File sessionFile)
 		
 		for(SampleSession sampleSession : session.samplePaths)
 		{
-			Sample sample = addSampleToList(treeViewHomePath + sampleSession.path);
+			Optional<Sample> sampleOptional = Optional.empty();
+			try {
+				sampleOptional = Optional.of(SPOperations.loadSample(treeViewHomePath + sampleSession.path));
```
Comment (markhalonen) [inline Viewer/src/net/relinc/viewer/GUI/HomeController.java]: @glenhalonen Here's an example of `Optional`

**Pull request merged**